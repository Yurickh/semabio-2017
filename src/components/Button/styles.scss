@import 'ambience/typography';
@import 'ambience/responsive';
@import 'ambience/color';

$simple-button-red: $color-blood;
$simple-button-green: $color-forest;

.simple-button {
  font-size: 18px;
  font-weight: $weight-bold;

  cursor: pointer;
  letter-spacing: 1px;

  color: #FFF;
  background-color: $simple-button-red;

  @include small-only {
    font-size: calc(14px + 1vw);

    width: 72vw;
  }

  &:hover {
    background-color: darken($simple-button-red, 3%);
  }

  &.-green {
    background-color: $simple-button-green;
  }

  &.-green:hover {
    background-color: darken($simple-button-green, 3%);
  }

  &.-outline {
    display: flex;
    padding: 15px 45px;

    color: $simple-button-green;
    border: 2px solid $simple-button-green;
    background-color: transparent;
  }

  &.-outline:hover {
    background-color: $simple-button-green;
    color: #FFF;
  }
}
