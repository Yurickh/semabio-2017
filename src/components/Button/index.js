import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class Button extends Component {
  render() {
    return (
      <button {...this.props} className={`simple-button -${this.props.color} ${this.props.className}`}>
        { this.props.children }
      </button>
    )
  }
}

Button.propTypes = {
  color: PropTypes.oneOf(['green', 'red', 'outline']),
  onClick: PropTypes.func,
}

Button.defaultProps = {
  color: 'red',
  onClick: _=>_,
}

export default Button
