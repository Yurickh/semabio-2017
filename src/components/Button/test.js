import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Button from '../Button'

describe('Button', () => {
  it('should render correctly', () => {
    const button = shallow(<Button />)
    expect(toJSON(button)).toMatchSnapshot()
  })
})
