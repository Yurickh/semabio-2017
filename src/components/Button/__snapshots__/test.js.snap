// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="simple-button -red  undefined"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;
